backoff.go

```diff
@@ -27,12 +27,34 @@ import (
 
 // DefaultBackoffConfig uses values specified for backoff in
 // https://github.com/grpc/grpc/blob/master/doc/connection-backoff.md.
+// Deprecated: use ConnectRetryParams instead.
 var DefaultBackoffConfig = BackoffConfig{
 	MaxDelay: 120 * time.Second,
 }
 
 // BackoffConfig defines the parameters for the default gRPC backoff strategy.
+// Deprecated: use ConnectRetryParams instead.
 type BackoffConfig struct {
 	// MaxDelay is the upper bound of backoff delay.
 	MaxDelay time.Duration
 }
+
+// ConnectRetryParams defines the parameters for the backoff during connection
```

Comment:
Bikeshedding time..

How about: "defines the parameters for connecting and retrying"?

And for the fields, make the ones related to backoff have "backoff in their name":

`BackoffBaseDelay`, `BackoffMultiplier`, `BackoffJitter`, `BackoffMaxDelay`.

Also, maybe s/Retry// from the name?

Reply:
Done.